README.md

```diff
@@ -117,6 +117,22 @@ pip install staramr
 
 However, you will have to install the external dependencies (listed below) separately.
 
+## Latest Code
```

Comment:
### General comments:


I found this article to be useful when I came from C# to Python; it helped me break into the Python way of doing things: http://dirtsimple.org/2004/12/python-is-not-java.html

Also [The Zen of Python](https://www.python.org/dev/peps/pep-0020/)

The [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html) has a lot of good suggestions for how to do things in Python "the Pythonic way". I like to do Python docstrings the Google way, but there are many other popular doc styles like the [NumPy way](https://numpydoc.readthedocs.io/en/latest/format.html). 

Hacking around in an IPython Notebook with some data to nail down a workflow or algorithm can be nice. It's something I do all the time since it provides very rapid feedback when I'm trying to do something new. There's really great support for all things Pandas, NumPy, matplotlib, etc in IPyNBs, and they render nicely on Github (e.g. https://github.com/phac-nml/refseq_masher/blob/master/ipynbs/get-refseq-taxonomic-info.ipynb).


Some nice features in Python/Python 3.6+:

- [optional type hinting](https://docs.python.org/3/library/typing.html) (`def foo(x: int, y: float, b: bool) -> Dict[str, Union[int, float, bool]]:`; you can use optional type hinting with Python 2.7, but it's not as seamless)
- [literal string interpolation](https://www.python.org/dev/peps/pep-0498/) (`s = f'Hello {person}, my name is {name}'`)
- [generators](https://wiki.python.org/moin/Generators)


### Python's [with](https://docs.python.org/3/reference/compound_stmts.html#with) statement

For example:

```python
with open('file.txt') as fh:
    for line in fh:
        do_something(line)
# file handle automatically closed no matter what :)
```

You can also use `with` with your own classes to clean stuff up like temporary files or close DB connections. You just need to specify the `__exit__()` method in your class.

### OO and handling data

Check out [`namedtuple`](https://docs.python.org/3/library/collections.html#collections.namedtuple) in `collections` built-in module for immutable heterogeneous collection data types (converted into a class where you can't modify any of the properties after instantiation). 

For OO in Python, I'd highly recommend [`attrs`](https://github.com/python-attrs/attrs) as it makes your classes much more readable and gives you a lot of nice things like a good `__repr__`, `__eq__`, etc, and easy dict conversion (`attr.asdict(your_object)`). Instead of `namedtuple`, I usually use `attrs`.


### Tabular BLAST + Pandas = awesome

I'm not sure if this would be appropriate for your use case, but I've found it quite nice and flexible to read tabular BLAST+ output with the specific output fields I'm interested in using `-outfmt "6 <list of output field names>"` straight into a Pandas DataFrame: 

```
df = pd.read_table('blast_output.b6', header=False, columns=BLAST_COLS)
df['coverage'] = df['length'] / df['qlen']
df_high_pid_cov = df[(df['pident'] > pid_threshold) & (df['coverage'] > cov_threshold)]
```

rather than using BioPython with BLAST+ XML output and various Python sorting/filtering/etc functions. 


### Constant lists from multiline strings

This is might seem slightly hacky, but for constant lists (like a list of BLAST+ output field names), I like to split on new-lines from a multiline string, e.g.

```
BLAST_COLS = '''
qseqid
stitle
pident
length
mismatch
gapopen
qstart
qend
sstart
send
evalue
bitscore
qlen
slen
sseq
'''.strip().split('\n')
```

It looks like you do it in `setup.py` for `classifiers`. 

I find it allows easy rearranging of column names for Pandas DataFrames so if I build a DataFrame from a dict then I can easily specify the order of columns I want.

```python
# df with columns in random/sorted order
df = pd.DataFrame(results_dict)
# df columns sorted according to list of columns COLS
df = df[COLS]
```

Reply:
Great to hear that you've found some of the info useful!

Good use of `with` statements :+1: 

For `attrs`, you can use the `@classmethod` decorator:
http://www.attrs.org/en/stable/init.html

The class var lists from multiline strings looks really nice and makes it clear what the columns and index are for each of the outputs. It's nice that all subclasses use `BlastResultsParser.parse_results()` now too.